src/compiler/checker.ts

```diff
@@ -9595,13 +9595,21 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     if (p.flags & SymbolFlags.Accessor && useAccessors) {
                         const result: AccessorDeclaration[] = [];
                         if (p.flags & SymbolFlags.SetAccessor) {
+                            const setter = p.declarations?.find((d): d is SignatureDeclaration => {
+                                if (d.kind === SyntaxKind.SetAccessor) {
+                                    return true;
+                                }
+                                const id = getNameOfDeclaration(d);
+                                return !!id && isIdentifier(id) && idText(id) === "set";
+                            });
+                            const setterSignature = setter && getSignatureFromDeclaration(setter);
                             result.push(setTextRange(factory.createSetAccessorDeclaration(
                                 factory.createModifiersFromModifierFlags(flag),
                                 name,
                                 [factory.createParameterDeclaration(
                                     /*modifiers*/ undefined,
                                     /*dotDotDotToken*/ undefined,
-                                    "arg",
+                                   (setterSignature?.parameters[0].valueDeclaration as BindingName | undefined) || "arg",

```

Comment:
Also, is this using the parameter's node directly? Probably, you need to be doing some cloning and such, e.g. what is done in `symbolToParameterDeclaration`. Noticing there that it has a bit of logic specifically for binding names, too.

Reply:
thanks for the tip, i reused `symbolToParameterDeclaration` here and added a test case to cover for what it fixes (stripping initializers from binding patterns)